src/transports/quic.rs

```diff
@@ -0,0 +1,180 @@
+}
+
+impl Transport for QuicTransport {
+    type OutConnectionConfig = QuicOutConnectionConfig;
+
+    fn start_listener(&mut self, address: SocketAddr) -> Result<(), PeerNetError> {
+        let mut poll = Poll::new().map_err(|err| PeerNetError::ListenerError(err.to_string()))?;
+        let mut events = Events::with_capacity(128);
+        let waker = Waker::new(poll.registry(), STOP_LISTENER)
+            .map_err(|err| PeerNetError::ListenerError(err.to_string()))?;
+        let listener_handle = std::thread::spawn({
+            let peer_db = self.peer_db.clone();
+            move || {
+                let mut socket = UdpSocket::bind(address)
+                    .expect(&format!("Can't bind TCP transport to address {}", address));
+                // Start listening for incoming connections.
+                poll.registry()
+                    .register(&mut socket, NEW_CONNECTION_SERVER, Interest::READABLE)
+                    .expect(&format!(
+                        "Can't register polling on TCP transport of address {}",
+                        address
+                    ));
+                let mut buf = [0; PUBLIC_KEY_SIZE_BYTES + SIGNATURE_SIZE_BYTES];
+                loop {
+                    println!("Listening for new connections...");
+                    // Poll Mio for events, blocking until we get an event.
+                    poll.poll(&mut events, None)
+                        .expect(&format!("Can't poll QUIC transport of address {}", address));
+
+                    // Process each event.
+                    for event in events.iter() {
+                        match event.token() {
+                            NEW_CONNECTION_SERVER => {
+                                //TODO: Error handling
+                                let (_num_recv, from_addr) = socket.recv_from(&mut buf).unwrap();
+                                let mut config =
+                                    quiche::Config::new(quiche::PROTOCOL_VERSION).unwrap();
+                                accept(
+                                    &ConnectionId::from_ref(&buf[..PUBLIC_KEY_SIZE_BYTES]),
+                                    None,
+                                    address,
+                                    from_addr,
+                                    &mut config,
+                                )
+                                .unwrap();
+                                let mut peer_db = peer_db.write();
+                                if peer_db.nb_in_connections < peer_db.config.max_in_connections {
+                                    peer_db.nb_in_connections += 1;
+                                    let peer = Peer::new(Endpoint {});
+                                    peer_db.peers.push(peer);
+                                }
+                            }
+                            STOP_LISTENER => {
+                                return;
+                            }
+                            _ => {}
```

Comment:
// We don't expect any events with tokens other than those we provided. (from mio doc)
                _ => unreachable!(),

Reply:
Done